echopype/mask/api.py

```diff
@@ -0,0 +1,306 @@
+
+    This function computes the frequency differencing as follows:
+    ``Sv_freqA - Sv_freqB operator diff``. Thus, if ``operator = "<"``
+    and ``diff = "5"`` the following would be calculated:
+    ``Sv_freqA - Sv_freqB < 5``.
+
+    Examples
+    --------
+    Compute frequency-differencing mask using a mock Dataset and channel selection:
+
+    >>> n = 5 # set the number of ping times and range samples
+    ...
+    >>> # create mock Sv data
+    >>> Sv_da = xr.DataArray(data=np.stack([np.arange(n**2).reshape(n,n), np.identity(n)]),
+    ...                      coords={"channel": ['chan1', 'chan2'],
+    ...                              "ping_time": np.arange(n), "range_sample":np.arange(n)})
+    ...
+    >>> # obtain mock frequency_nominal data
+    >>> freq_nom = xr.DataArray(data=np.array([1.0, 2.0]),
+    ...                         coords={"channel": ['chan1', 'chan2']})
+    ...
+    >>> # construct mock Sv Dataset
+    >>> Sv_ds = xr.Dataset(data_vars={"Sv": Sv_da, "frequency_nominal": freq_nom})
+    ...
+    >>> # compute frequency-differencing mask using channel names
+    >>> echopype.mask.frequency_difference(source_Sv=mock_Sv_ds, storage_options={}, freqAB=None,
+    ...                                    chanAB = ['chan1', 'chan2'],
+    ...                                    operator = ">=", diff=10.0)
+    <xarray.DataArray 'frequency_difference_mask' (ping_time: 5, range_sample: 5)>
+    array([[False, False, False, False, False],
+           [False, False, False, False, False],
+           [ True,  True,  True,  True,  True],
+           [ True,  True,  True,  True,  True],
+           [ True,  True,  True,  True,  True]])
+    Coordinates:
+      * ping_time     (ping_time) int64 0 1 2 3 4
+      * range_sample  (range_sample) int64 0 1 2 3 4
+    """
+
+    # check that non-data related inputs were correctly provided
+    _check_freq_diff_non_data_inputs(freqAB, chanAB, operator, diff)
+
+    # check the source_Sv input
+    source_Sv = _check_source_Sv_freq_diff(source_Sv, storage_options, freqAB, chanAB)
+
+    # determine chanA and chanB
+    if freqAB is not None:
+
+        # obtain position of frequency provided in frequency_nominal
+        freqA_pos = np.argwhere(source_Sv.frequency_nominal.values == freqAB[0]).flatten()[0]
+        freqB_pos = np.argwhere(source_Sv.frequency_nominal.values == freqAB[1]).flatten()[0]
+
+        # get channel corresponding to frequency provided
+        chanA = source_Sv.channel.isel(channel=freqA_pos)
+        chanB = source_Sv.channel.isel(channel=freqB_pos)
+
+    else:
+        # get individual channels
+        chanA = chanAB[0]
+        chanB = chanAB[1]
+
+    # get the left-hand side of condition
+    lhs = source_Sv["Sv"].sel(channel=chanA) - source_Sv["Sv"].sel(channel=chanB)
+
+    # create mask using operator lookup table
+    da = xr.where(str2ops[operator](lhs, diff), True, False)
+
+    # assign a name to DataArray
+    da.name = "frequency_difference_mask"
```

Comment:
I think we can just call this mask right now and leave the other parts to the provenance that should be recorded in attributes (i.e., this mask is created using freq diff function).
```suggestion
    # assign a name to DataArray
    da.name = "mask"
```